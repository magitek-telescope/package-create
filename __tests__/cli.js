const sao = require('sao')
const path = require('path')

const template = path.join(__dirname, '../')

const getPackageFields = stream => {
  pkg = JSON.parse(stream.fileContents('package.json'))
  delete pkg.name
  delete pkg.author
  delete pkg.version
  delete pkg.description
  return pkg
}

const createStream = async (answers) => (await sao.mockPrompt(template, answers))

describe('cli.js', async () => {
  test('defaults', async () => {
    const stream = await createStream()
    expect(stream.fileList).toMatchSnapshot()
    expect(getPackageFields(stream)).toMatchSnapshot()
  })

  describe('use Babel', async () => {
    test('defaults', async () => {
      const stream = await createStream()
      expect(stream.fileList).toMatchSnapshot()
      expect(getPackageFields(stream)).toMatchSnapshot()
    })
  })

  describe('use TypeScript', async () => {
    test('defaults', async () => {
      const stream = await createStream({ language: 'TypeScript' })
      expect(stream.fileList).toMatchSnapshot()
      expect(getPackageFields(stream)).toMatchSnapshot()
    })
  })
})
