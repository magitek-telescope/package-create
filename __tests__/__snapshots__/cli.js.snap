// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cli.js defaults 1`] = `
Array [
  ".babelrc",
  ".circleci/config.yml",
  ".gitignore",
  "package.json",
  "spec/src/index.spec.js",
  "src/index.js",
]
`;

exports[`cli.js defaults 2`] = `
Object {
  "Author": "HANATANI Takuma",
  "devDependencies": Object {
    "babel-cli": "^6.26.0",
    "babel-core": "^6.26.3",
    "babel-jest": "^23.4.0",
    "babel-plugin-add-module-exports": "^0.2.1",
    "babel-plugin-transform-class-properties": "^6.24.1",
    "babel-preset-env": "^1.7.0",
    "jest": "^23.4.1",
    "prettier": "^1.13.7",
  },
  "jest": Object {
    "transform": Object {
      "^.+\\\\.jsx?$": "babel-jest",
    },
  },
  "license": "MIT",
  "main": "lib/index.js",
  "scripts": Object {
    "build": "babel './src/**/*.js' --out-dir ./lib/",
    "dev": "yarn build -w",
    "format": "prettier './{src, spec}/**/*.js' --write",
    "test": "jest",
    "test:coverage": "jest --coverage",
  },
}
`;

exports[`cli.js use Babel defaults 1`] = `
Array [
  ".babelrc",
  ".circleci/config.yml",
  ".gitignore",
  "package.json",
  "spec/src/index.spec.js",
  "src/index.js",
]
`;

exports[`cli.js use Babel defaults 2`] = `
Object {
  "Author": "HANATANI Takuma",
  "devDependencies": Object {
    "babel-cli": "^6.26.0",
    "babel-core": "^6.26.3",
    "babel-jest": "^23.4.0",
    "babel-plugin-add-module-exports": "^0.2.1",
    "babel-plugin-transform-class-properties": "^6.24.1",
    "babel-preset-env": "^1.7.0",
    "jest": "^23.4.1",
    "prettier": "^1.13.7",
  },
  "jest": Object {
    "transform": Object {
      "^.+\\\\.jsx?$": "babel-jest",
    },
  },
  "license": "MIT",
  "main": "lib/index.js",
  "scripts": Object {
    "build": "babel './src/**/*.js' --out-dir ./lib/",
    "dev": "yarn build -w",
    "format": "prettier './{src, spec}/**/*.js' --write",
    "test": "jest",
    "test:coverage": "jest --coverage",
  },
}
`;

exports[`cli.js use TypeScript defaults 1`] = `
Array [
  ".circleci/config.yml",
  ".gitignore",
  "package.json",
  "spec/src/index.spec.ts",
  "src/index.ts",
  "tsconfig.json",
]
`;

exports[`cli.js use TypeScript defaults 2`] = `
Object {
  "Author": "HANATANI Takuma",
  "devDependencies": Object {
    "@types/jest": "^23.3.0",
    "@types/node": "^10.5.2",
    "jest": "^23.4.1",
    "prettier": "^1.13.7",
    "ts-jest": "^23.0.1",
    "typescript": "^2.9.2",
  },
  "jest": Object {
    "moduleFileExtensions": Array [
      "ts",
      "tsx",
      "js",
      "jsx",
      "json",
      "node",
    ],
    "testRegex": "(/__tests__/.*|(\\\\.|/)(test|spec))\\\\.ts$",
    "transform": Object {
      "^.+\\\\.tsx?$": "ts-jest",
    },
  },
  "license": "MIT",
  "main": "lib/index.js",
  "scripts": Object {
    "build": "tsc",
    "dev": "yarn build -w",
    "format": "prettier './{src, spec}/**/*.{js,ts}' --write",
    "test": "jest",
    "test:coverage": "jest --coverage",
  },
}
`;
